ruby/ql/lib/codeql/ruby/ApiGraphs.qll

```diff
@@ -1096,13 +1096,13 @@ module API {
       /** The method accessed at `call`, synthetically treated as a separate object. */
       MkMethodAccessNode(DataFlow::CallNode call) or
       /** The module object `mod` with epsilon edges to its ancestors. */
-      MkModuleObjectUp(DataFlow::ModuleNode mod) or
+      MkModuleObjectUp(DataFlow::ModuleNode mod) { not mod.getQualifiedName() = "Object" } or
       /** The module object `mod` with epsilon edges to its descendents. */
-      MkModuleObjectDown(DataFlow::ModuleNode mod) or
+      MkModuleObjectDown(DataFlow::ModuleNode mod) { not mod.getQualifiedName() = "Object" } or
       /** Instances of `mod` with epsilon edges to its ancestors. */
-      MkModuleInstanceUp(DataFlow::ModuleNode mod) or
+      MkModuleInstanceUp(DataFlow::ModuleNode mod) { not mod.getQualifiedName() = "Object" } or
       /** Instances of `mod` with epsilon edges to its descendents, and to its upward node. */
-      MkModuleInstanceDown(DataFlow::ModuleNode mod) or
```

Comment:
Does it make sense to create an abstraction/class/predicate to capture this repeated logic?

Reply:
It would conflict with some of the follow-up changes I mentioned in my update so I decided to just keep it simple.